docs/parser_plugins.md

```diff
@@ -1 +1,264 @@
+Normally, you don't have to touch anything there for NOMAD to pick up on your parser.
+It does allows you to control the loading of plugins and their options.
+Closing the "dependence" cycle, `myparser.py` has to read in these options.
+
+## Fundamentals of Parsing
+
+As parsing involves the mapping between two well-defined formats, one could expect it to be trivial.
+In practice, however, parser developers have to manage discrepancies in semantics, shape, data type or units.
+This has lead to five distinct categories of responsibility for the developer to manage.
+Here, they are ordered to match the parser's execution:
+
+1. **file selection** - navigate the upload's folder structure and select the relevant files.
+2. **source extraction** - read the files into Python. This step may already include some level of data field filtering.
+3. **source to target** - map the data of interest with their counterparts in the target/NOMAD schema. This is where the bulk of the filtering happens.
+4. **data mangling** - manipulate the data to match the target/NOMAD `Quantity`s' specification, e.g. dimensionality, shape. This may include computing derived properties not present in the original source files.
+5. **archive construction** - build up a Python `EntryArchive` object using the classes provided by the target/NOMAD schema. NOMAD will automatically write it commit it to the database as an `archive.json`
+
+Blurring these responsibilities leads to a wild-growth in parser design and added complexity, especially in larger, more feature-rich parsers.
+NOMAD therefore offers powerful _tools_ and documentation on _best practices_ to help the parser developer manage each distinct responsibility.
+The exact solutions are, in the same order:
+
+1. `MatchingParser` - This class passes the file contents on to the parser class, which extends it via inheritance. Since it interfaces with the NOMAD base directly, it will read most of the settings automatically in from there. Options may be tweaked via the entry-steps mechanism. 
+2. `XMLParser` and co. / `TextParser` - There are several _reader classes_ for loading common source formats into Python data types. Examples include the `XMLParser` and `HDF5Parser`. We will demonstrate `XMLParser` in [[Parsing Hierarchical Tree Formats]]. Plain text files, meanwhile, involve an additional _matching step_ via the `TextParser`. More on this in [[From Text to Hierarchies]].
+3. `MappingAnnotationModel` - This is arguably the most involved part for the parser developer, as this is where the external data gets further semantically enriched and standardized. It requires doamin expertise to understand the relationship between the data fields in the source file and the [NOMAD-Simulations schema](nomad_simulations.md). If step 2 went well, this step only involves _annotating_ the target schema via `MappingAnnotationModel`. More on this in [[Parsing Hierarchical Tree Formats]].
```

Comment:
typo "domain"

Reply:
"...via the `MappingAnnotationModel` class"

Reply:
I think there is a little bit missing here, including possible a disclaimer about this class and what the alternative was/is:

"...NOMAD-Simulations schema. This can be achieved through a "manual" population of each piece of data into the NOMAD `EntryArchive` object. However, this can be tedious and often leads to complex parser structures that are difficult to extend. For this reason, the `MappingAnnotationModel` class is currently being developed to automatically perform the necessary data mapping through simple _annotations_. More on this..."

Just a first suggestion, you likely need to adjust it